types/express-http-proxy/index.d.ts

```diff
@@ -12,10 +13,43 @@ import { Response } from "express-serve-static-core";
 
 interface ProxyOptions {
     proxyReqPathResolver?: (req: Request) => string;
-    proxyReqOptDecorator?: (proxyReqOpts: RequestOptions, srcReq: Request) => RequestOptions;
-    userResHeaderDecorator?: (headers: IncomingHttpHeaders, userReq: Request, userRes: Response, proxyReq: Request, proxyRes: Response) => OutgoingHttpHeaders;
+    proxyReqOptDecorator?: (
+        proxyReqOpts: RequestOptions,
+        srcReq: Request
+    ) => RequestOptions;
+    userResHeaderDecorator?: (
+        headers: IncomingHttpHeaders,
+        userReq: Request,
+        userRes: Response,
+        proxyReq: Request,
+        proxyRes: Response
+    ) => OutgoingHttpHeaders;
+    filter?: (req: Request, res: Response) => boolean;
+    userResDecorator?: (
+        proxyRes: Response,
+        proxyResData: JSON,
```

Comment:
The `JSON` type is wrong here (it refers to the `JSON` object holding `JSON.parse`, ...; it's not about "some JSON data").

I conclude from the [sample in the docs](https://github.com/villadora/express-http-proxy#userresdecorator-was-intercept-supports-promise) that it can be any payload data (like a Buffer or JSON object), ... so I would propose the type `{} | null | undefined` (and in later TypeScript versions the type `unknown`).